config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -3039,7 +3040,12 @@ public LogoutSpec logoutHandler(ServerLogoutHandler logoutHandler) {
 			return addLogoutHandler(logoutHandler);
 		}
 
-		private LogoutSpec addLogoutHandler(ServerLogoutHandler logoutHandler) {
+		/**
+		 * Adds a logout handler in the last position.
+		 * @param logoutHandler
+		 * @return the {@link LogoutSpec} to configure
+		 */
+		public LogoutSpec addLogoutHandler(ServerLogoutHandler logoutHandler) {
```

Comment:
Instead of exposing `addLogoutHandler`, would `logout(Consumer<List<ServerLogoutHandler>> consumer)` also service your needs? The reason this is nice it because it also allows you to remove values. Please see `OneTimeTokenLogoutSpec#authenticationSuccessHandler` for an example.